### PR TITLE
Database Model

### PR DESCRIPTION
**Pull request by 24neurons**
# Description

PR type:
- New features

## What is the purpose of this pull request?

Add User, Problem, Contest peewee model to database.

## How has this pull request been tested?

Locally tested and circular import bug fixed.

### CODE REVIEW

**Comment by hung3a8** [comment]
Next time, consider using a more detailed name for the PRs. For example, instead of "Database", please call it "Database models" to distinguish from other database-related PRs.

File: server/db/user.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    last_name = peewee.CharField(null=True)
+
+    # Join date of a user
+    join_date = peewee.DateTimeField()
+
+    # Last visit
+    last_visit_date = peewee.DateTimeField(default=datetime.datetime.now())
+
+    # Determine if this user is admin or not
+    is_admin = peewee.BooleanField(default=False)
```
Comment (hung3a8) [inline server/db/user.py:32]: Please use `Field.help_text` instead of commenting.

File: server/db/contest.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    contest_id = peewee.IntegerField(primary_key=True, help_text='Auto increment primary key')
+
+    contest_code = peewee.CharField(unique=True, help_text='Code name of contest')
+
+    contest_description = peewee.TextField(help_text='Description of contest')
+
+    contest_start = peewee.DateTimeField(formats='%Y-%m-%d %H:%M:%S', default=datetime.datetime(2010, 1, 1, 0, 0, 0),
+                                         help_text='Beginning time of contest')
+    contest_end = peewee.DateTimeField(formats='%Y-%m-%d %H:%M:%S', null=True, help_text='Ending time of contest')
+
```
Comment (hung3a8) [inline server/db/contest.py:21]: 1. Remove the `contest_` prefix. Accessing the field with `contest.id` is much simpler than `contest.contest_id`.
2. Remove the blank lines. Only use the blank lines to separate groups of variables like "contest time" or "contest info".

File: server/db/contest.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    if database.__class__ == peewee.MySQLDatabase:
+        contest_settings = playhouse.mysql_ext.JSONField(help_text='Contest settings in JSON')
+    else:
+        contest_settings = playhouse.sqlite_ext.JSONField(help_text='Contest settings in JSON')
```
Comment (hung3a8) [inline server/db/contest.py:25]: You can import `JSONFIeld` in `__init__.py` like this:
```python
if config.database['mysql']:
    from playhouse.mysql_ext import JSONField
elif config.database['sqlite']:
    from playhouse.sqlite_ext import JSONField
```
When you have to use `JSONField`, you just have to do this:
```python
from server.db import JSONField
```
The class will be imported correctly without performing any further class checkups like this.

**Comment by hung3a8** [comment]
Also, A script is required to initialize the database, as using the interactive console just to initialize the database is kinda annoying. Tell me if you need some help with that :3

File: server/db/contest.py (hunk 2 above)
Comment (24neurons) [reply server/db/contest.py]: Finished

**Comment by 24neurons** [comment]
Field.help_text formulated, please review and leave comments whether it needs further amendments

File: server/db/contest.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    start = peewee.DateTimeField(formats='%Y-%m-%d %H:%M:%S',
+                                 help_text=('Help on Contest start key\n'
+                                            ' Field:\n    peewee.DateTimeField\n'
+                                            ' Description:\n'
+                                            '    Beggining time with format year-month-day, hour-minute-second'))
+    end = peewee.DateTimeField(formats='%Y-%m-%d %H:%M:%S', null=True,
+                               help_text=('Help on Contest end key\n'
+                                          ' Field:\n    peewee.DateTimeField\n'
+                                          ' Description:\n'
+                                          '    Ending time with format year-month-day, hour-minute-second, null=True'))
```
Comment (hung3a8) [inline server/db/contest.py:30]: There's no need to write everything down this detailed xD.

File: server/db/contest.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import peewee
+
+from server.db import JSONField, database
+
+
+class Contest(peewee.Model):
+
+    id = peewee.IntegerField(primary_key=True, help_text=('Help on Contest id key\n'
```
Comment (hung3a8) [inline server/db/contest.py:8]: `peewee` automatically creates an `id` field as a PK. Check [this](https://docs.peewee-orm.com/en/latest/peewee/models.html#primary-keys-composite-keys-and-other-tricks) for more information.

**Review by hung3a8 — APPROVED** [review]
LGTM rn. The manager can be implemented later. Great work :tada:!

**Pull request merged**